frontend/src/routes/Router.tsx

```diff
@@ -11,35 +11,35 @@ import {
 } from "react-router-dom";
 
 const routes = createRoutesFromElements(
-  // element={<Layout />}
   <Route>
+    {/* Unauthenticated landing: */}
     <Route index element={<Landing />} />
-
-    {/* Protect all other routes based on authentication */}
-    <Route element={<AuthenticationGuard />}></Route>
-
-    {/* Login page in case unauthenticated */}
-    <Route
-      element={
-        <AuthenticationGuard
-          guardType="unauthenticated"
-          redirectPath="/login"
-        />
-      }
-    >
-      <Route path="login" element={<LoginForm />} />
+    <Route path="/" element={<NavigationBar />}>
+      <Route
+        element={
+          <AuthenticationGuard
+            guardType="unauthenticated"
+            redirectPath="/login"
+          />
+        }
+      >
+        <Route path="login" element={<LoginForm />} />
+      </Route>
+      <Route
+        element={
+          <AuthenticationGuard
+            guardType="unauthenticated"
+            redirectPath="/register"
+          />
+        }
+      >
+        <Route path="register" element={<RegisterForm />} />
+      </Route>
     </Route>
```

Comment:
The router should render the overall views and not the components that the views consist of. In other words, right now this would suggest that the base route renders the navigationBar component. Rather, the navigationbar should be inside the component that the route renders here.

Moreover, shouldn't the navigation be visible for each page?

Reply:
And to clarify, we could have a layout similar to this:

```javascript
  <Route path="/" element={<MainLayout />}>
     {/* Other routes */}
  </Route>
 ```
 
 The navigation bar along with other components that are consistent throughout the application could then be placed inside the layout.